### PR TITLE
Change banner alert to snackbar

### PR DESCRIPTION
**Pull request by dfguerrerom**
I've changed the App `banner` to a `v.Snackbar`, this will prevent the behavior commented on #438. 
Although @12rambau pointed out that would be nice to use the `timeout` parameter, I've set it as 0 (permanent) as default, mainly because —unless the developer wants the opposite, we might be sure the user has read the message.

image

what do you think?

### CODE REVIEW

**Comment by dfguerrerom** [comment]
Doing this, I've realized that would be nice to have a new method in `SepalWidget` to search and return a specific children's widget based on an optional `id` parameter... something like `get_children(id)`, what do you think?

**Comment by 12rambau** [comment]
> Doing this, I've realized that would be nice to have a new method in `SepalWidget` to search and return a specific children's widget based on an optional `id` parameter... something like `get_children(id)`, what do you think?

aggreed but I would do it in another PR. Ithink both `get_children(index)` and `set_children(index)` could be useful

**Comment by 12rambau** [comment]
I love the idea but I'm afraid the nested functions will lead to unwanted complexity. 

Within `sepal-ui`, no one will ever use a `Snackbar` outside of the `App` object. I would say let's overwrite the Snackbar object (as we overwrite the `Alert` one) and add an extra parameter for timeout (true or false) and compute the time the banner is displayed using this [code](https://ux.stackexchange.com/questions/22520/how-long-does-it-take-to-read-x-number-of-characters/61976#61976). what do you think ?

**Comment by dfguerrerom** [comment]
> I love the idea but I'm afraid the nested functions will lead to unwanted complexity.
> 
> Within `sepal-ui`, no one will ever use a `Snackbar` outside of the `App` object. I would say let's overwrite the Snackbar object (as we overwrite the `Alert` one) and add an extra parameter for timeout (true or false) and compute the time the banner is displayed using this [code](https://ux.stackexchange.com/questions/22520/how-long-does-it-take-to-read-x-number-of-characters/61976#61976). what do you think ?

Yes, I agreed. In this way is cleaner.
Instead of overriding the Snackbar vuetify widget I've created a `Banner`, what do you think?

**Comment by 12rambau** [comment]
> Yes, I agreed. In this way is cleaner. Instead of overriding the Snackbar vuetify widget I've created a `Banner`, what do you think?

sounds perfect

File: sepal_ui/sepalwidgets/alert.py [hunk 1]
```diff
@@ -359,43 +358,45 @@ def add_msg(self, msg, loading=False):
...
-            type_ = kwargs["type"]
+        # the z-index value of the nav drawer when the screen is small
+        drawer_level = 6
 
-        if type_ not in TYPES:
-            raise ValueError(
-                f"type {type_} is not a valid type. Available types are: {TYPES}"
-            )
+        # compute the type and default to "info" if it's not existing
+        type_ = type_ if (type_ in TYPES) else TYPES[0]
```
[the remark this answers is not in the document]
Comment (dfguerrerom) [reply sepal_ui/sepalwidgets/alert.py:381]: I think now is looking better.
I've been messing around with the buttons and there are two un-desired behaviors:
- As we are using the `nb_banner` to count the number of displayed banners and also to control the overlay between them, when you close and open them, the `z-index` of the previous one is hidden in the background, so the last open banner is not backing to the front. Perhaps we can fix it by adding a new variable only to control the `z-index`

test

[the remark this answers is not in the document]
Comment (12rambau) [reply sepal_ui/sepalwidgets/alert.py:381]: ok then let's make it consistent throughout the file and apply the same behavior to other components that use the `TYPE` variable

**Comment by dfguerrerom** [comment]
I think now is looking better.
I've been messing around with the buttons and there are two un-desired behaviors:
- As we are using the `nb_banner` to count the number of displayed banners and also to control the overlay between them, when you close and open them, the `z-index` of the previous one is hidden in the background, so the last open banner is not backing to the front. Perhaps we can fix it by adding a new variable only to control the `z-index`?

test

File: sepal_ui/sepalwidgets/alert.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (12rambau) [reply sepal_ui/sepalwidgets/alert.py]: solved in https://github.com/12rambau/sepal_ui/pull/442/commits/767b607d3d1b3ab4ffc112e1fa651ea465183584

**Comment by 12rambau** [comment]
> * As we are using the `nb_banner` to count the number of displayed banners and also to control the overlay between them, when you close and open them, the `z-index` of the previous one is hidden in the background, so the last open banner is not backing to the front. Perhaps we can fix it by adding a new variable only to control the `z-index`?

solved in https://github.com/12rambau/sepal_ui/pull/442/commits/0f915912e397802543a5bf63e5c5eced155e30a8

now only the oldest one is displayed and the other are queued in the background. every time you click on "close" sepal_ui will try to display a newer (if existing).

**Review by 12rambau — APPROVED** [review]

**Pull request merged**